src/plugins/intel_cpu/src/shape_inference/custom/fullyconnected.cpp

```diff
@@ -15,15 +15,16 @@ Result FCShapeInfer::infer(
     const VectorDims& activationShape = input_shapes[0].get();
     const VectorDims& weightShape = input_shapes[1].get();
     size_t activationRank = activationShape.size();
-    size_t channelRank = weightShape.size() - 1;
+    // size_t channelRank = weightShape.size() - 1;
+    size_t channelRank = 1;
 
     // activation   weight    output_shape
     // NCHW         CoCHW     NCo
     // TNC          CoC       TNCo
     // NC           CoC       NCo
     VectorDims outputShape(out_rank, 1);
     // set Co
-    outputShape.back() = weightShape[0];
+    outputShape.back() = std::accumulate(weightShape.begin(), weightShape.end() - 1, 1, std::multiplies<Dim>());
```

Comment:
I am wondering does FullyConnected::validate_and_infer_types() always return 2D shape on the output?

Reply:
Let's resolve this when we implement an actual validate_an_infer_types for FullyConnected Op